src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/BatchEmitter.java

```diff
@@ -379,6 +410,7 @@ private Runnable getPostRequestRunnable(int numberOfEvents) {
                 LOGGER.error("BatchEmitter event sending error: {}", e.getMessage());
                 if (batchedEvents != null) {
                     eventStore.cleanupAfterSendingAttempt(true, batchedEvents.getBatchId());
+                    callback.onFailure(FailureType.CONNECTION_FAILURE, true, new ArrayList<>(batchedEvents.getPayloads()));
```

Comment:
I think the connection failure error is represented as the -1 status code returned from the HTTP client, see the [OkHttpClientAdapter implementation](https://github.com/snowplow/snowplow-java-tracker/blob/issue/339-callbacks/src/main/java/com/snowplowanalytics/snowplow/tracker/http/OkHttpClientAdapter.java#L122). Perhaps we should check for that case among the if cases above?

I'm not actually sure what Exception is caught in this catch clause – I tried to remove it and nothing complained, all tests passed. Do you have any idea what is it for? If we could make the failure type more specific to the problem, that would be great.

Reply:
Most likely InterruptedException? I couldn't work out how to test it